.azure/pipeline/azure-cicd.yml

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+#################################################################################
+#                      OneBranch Pipelines - PR Build                           #
+# This pipeline was created by EasyStart from a sample located at:              #
+#   https://aka.ms/obpipelines/easystart/samples                                #
+# Documentation:  https://aka.ms/obpipelines                                    #
+# Yaml Schema:    https://aka.ms/obpipelines/yaml/schema                        #
+# Retail Tasks:   https://aka.ms/obpipelines/tasks                              #
+# Support:        https://aka.ms/onebranchsup                                   #
+#################################################################################
+
+# https://aka.ms/obpipelines/triggers
+trigger:
+  ## disable batching of changes when a push happens.
+  batch: false
+  branches:
+    include:
+    - main
+    - release/*
+
+pr:
+- main
+- release/*
+
+schedules:
+- cron: '0 21 * * *'
+  displayName: Schedule CI/CD
+  branches:
+    include:
+    - main
+    - release/*
+  always: true
+
+parameters: # parameters are shown up in ADO UI in a build queue time
+- name: 'debug'
+  displayName: 'Enable debug output'
+  type: boolean
+  default: false
+
+variables:
+  CDP_DEFINITION_BUILD_COUNT: $[counter('', 0)] # needed for onebranch.pipeline.version task https://aka.ms/obpipelines/versioning
+  system.debug: ${{ parameters.debug }}
+  ENABLE_PRS_DELAYSIGN: 0
+  ROOT: $(Build.SourcesDirectory)
+  REPOROOT: $(Build.SourcesDirectory)
+  OUTPUTROOT: $(REPOROOT)\out
+  NUGET_XMLDOC_MODE: none
+
+  WindowsContainerImage: 'cdpxwin1809.azurecr.io/global/vse2022:latest' # Docker image which is used to build the project https://aka.ms/obpipelines/containers
+
+jobs:
+
+# regular
+# Always run this job.
+- ${{ if in(variables['Build.Reason'], 'PullRequest', 'BatchedCI', 'IndividualCI', 'Schedule') }}:
+  - template: ./azure-reusable-build.yml
+    parameters:
+      name: regular
+      build_artifact: Build-x64
+      generate_release_package: true
+      build_nuget: true
+      build_options: /p:ReleaseJIT=True
+
+# api_test
+# Always run this job.
+- ${{ if in(variables['Build.Reason'], 'PullRequest', 'BatchedCI', 'IndividualCI', 'Schedule') }}:
+  - template: ./azure-reusable-test.yml
+    parameters:
+      name: api_test
+      test_command: '.\api_test.exe'
```

Comment:
While this is a draft, it'll be good to sort out if are we getting any benefits in scalability or reduced PR merge time if we onboard just some of the tests that are currently running on the self-hosted runners.

Since we don't want to reduce coverage, we likely don't want to disable the self-hosted runners until we can onboard all the tests they run.

At the same time, having the Azure runners execute some of the tests in parallel to the self-hosted runners seems unenecessary.

Reply:
If we really wanted to do a partial onboarding, then I'd at least suggest removing the api_test fro the self-hosted runner script to avoid using resources twice.

Reply:
@rectified95 in the interest of making some forward progress, let us onboad the single machine tests first. Then go for the more difficult 2 machine tests. However, lets remove the single machine tests from the existing selfhost-runner based tests to avoid duplication.

Reply:
@shankarseal Note this is also about KM dump collection, which the Azure nodes won't have until we set up the 1ES pool.